compiler/src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -107,8 +107,9 @@ trait TypeAssigner {
     val tpe1 = accessibleType(tpe, superAccess)
     if tpe1.exists then tpe1
     else tpe match
-      case tpe: NamedType => inaccessibleErrorType(tpe, superAccess, pos)
-      case NoType => tpe
+      case tpe: NamedType if !tpe.termSymbol.hasPublicInBinary =>
```

Comment:
Since this only makes sense when typechecking inlined code, could we check if this is inside an inlined tree? If this is too tricky to check, maybe just add a comment.

Reply:
Done